src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -953,15 +956,35 @@ public Optional<File> findRapidScanConfig() throws OperationException {
                 .findRapidScanConfig(directoryManager.getSourceDirectory())
         );
     }
-    
+
+    private int countBdioEntryFiles() throws IntegrationException {
+        File bdioFile = this.fileFinder.findFile(directoryManager.getBdioOutputDirectory(), "*.bdio");
+        Bdio2ContentExtractor bdio2Extractor = new Bdio2ContentExtractor();
+        return bdio2Extractor.extractContent(bdioFile).size() - 1; // excludes bdio header file
```

Comment:
Header count acts as natural +1. Should not exclude it.

Reply:
Yeah I see how you're looking at it. 
This method was meant to only count the entry files (i.e. N from the design doc) and the return value is indexing the fibonacci sequence array correctly. Example: 3 entries -> 4th number in the array -> value is 2. 
I did it this way to make it more in sync with the design doc logic shared with everyone so it's easy to follow.
I kept the method name generic for future reusability. 
Most of the other suggested changes around this make sense if we change the method's meaning to "getting fibonacci index". Happy to update the flow if you still think this is a better way